src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -66,22 +69,67 @@
         return getAcceptableTokens();
     }
 
+    /**
+     * Whether to allow default keyword not in last but surrounded with case.
+     * @param newValue whether to ignore checking.
+     */
+    public void setskipIfLastAndSharedWithCase(boolean newValue) {
+        skipIfLastAndSharedWithCase = newValue;
+    }
+
     @Override
     public void visitToken(DetailAST ast) {
         final DetailAST defaultGroupAST = ast.getParent();
         //default keywords used in annotations too - not what we're
         //interested in
         if (defaultGroupAST.getType() != TokenTypes.ANNOTATION_FIELD_DEF
                 && defaultGroupAST.getType() != TokenTypes.MODIFIERS) {
-            final DetailAST switchAST = defaultGroupAST.getParent();
-            final DetailAST lastGroupAST =
-                switchAST.getLastChild().getPreviousSibling();
 
-            if (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
-                || defaultGroupAST.getColumnNo()
-                    != lastGroupAST.getColumnNo()) {
+            boolean containsInLastGroup = ast.getPreviousSibling() == null;
+            if (!skipIfLastAndSharedWithCase
+                    || containsInLastGroup) {
+                containsInLastGroup = containsInLastGroup(ast.getParent());
+            }
+            final boolean containsCase = containsCaseInBlock(ast);
+            if (containsCase || containsInLastGroup) {
                 log(ast, MSG_KEY);
             }
         }
     }
+
+    /**
+     * Determines after default case group any case group is there.
+     *
+     * @param ast root node.
+     * @return true if contains any case group after default block or false.
+     */
+    private boolean containsInLastGroup(DetailAST ast) {
+        boolean containsInLastGroup = false;
+        DetailAST root = ast.getNextSibling();
+        while (root != null) {
+            if (root.getType() == TokenTypes.CASE_GROUP) {
+                containsInLastGroup = true;
```

Comment:
If we have found the answer we are looking for, then there is no reason to keep going through the loop.
add ` break` right after here.